### PR TITLE
Fixes to run nightly on macsec enabled ports

### PR DESCRIPTION
**Pull request by judyjoseph**
Summary:
Plan is to run the entire sonic-mgmt test pipeline on testbeds where macsec is enabled.


### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?
```
jujoseph@sonic-mgmt-jujoseph_master:/var/src/sonic-mgmt-int/tests$ ./run_tests.sh -c decap/test_decap.py -i '../ansible/strsvc,../ansible/veos' -n vmsvc3-t2-xxxx-1  -t 't2,any' -u -e "--skip_sanity --disable_loganalyzer  --enable_macsec --macsec_profile=MACSEC_PROFILE" 
=== Running tests in groups ===
Running: python2 -m pytest decap/test_decap.py --inventory ../ansible/strsvc,../ansible/veos --host-pattern svcstr-xxxx-lc1-1,svcstr-xxxx-lc2-1,svcstr-xxxx-lc3-1,svcstr-xxxx-sup-1 --testbed vmsvc3-t2-xxxx-1 --testbed_file /var/src/sonic-mgmt-int/ansible/testbed.yaml --log-cli-level warning --log-file-level debug --kube_master unset --showlocals --assert plain --show-capture no -rav --allow_recover --ignore=ptftests --ignore=acstests --ignore=saitests --ignore=scripts --ignore=k8s --ignore=sai_qualify --junit-xml=logs/tr.xml --log-file=logs/test.log --topology t2,any --skip_sanity --disable_loganalyzer --enable_macsec --macsec_profile=MACSEC_PROFILE
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
========================================================================================================== test session starts ==========================================================================================================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1
ansible: 2.8.20
rootdir: /var/src/sonic-mgmt-int/tests, inifile: pytest.ini
plugins: ansible-2.2.4, celery-4.4.7, html-1.22.1, repeat-0.9.1, forked-1.3.0, metadata-1.11.0, xdist-1.28.0, allure-pytest-2.8.22
collecting ... 
---------------------------------------------------------------------------------------------------------- live log collection ----------------------------------------------------------------------------------------------------------
collected 4 items                                                                                                                                                                                                                       

decap/test_decap.py::test_decap[MACSEC_PROFILE-ttl=pipe, dscp=uniform, vxlan=disable] PASSED                                                                                                                                      [ 25%]
```
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/macsec/macsec_config_helper.py
Comment (Pterosaur) [inline tests/macsec/macsec_config_helper.py:1]: Why are these sleep calls moved from the action functions to the submit_task function?

Comment (judyjoseph) [reply tests/macsec/macsec_config_helper.py:1]: I found the sleep was for 6 secs in each of macsec enable and disable which was not required. Instead put a small wait after each macsec enable disable to make sure session is established. 
Have tried multiple times with 32 macsec sessions -- and sessions came up consistent

**Review by arlakshm — DISMISSED** [review]

**Comment by judyjoseph** [comment]
Resolved conflicts with master @arlakshm to review again/approve

**Review by arlakshm — APPROVED** [review]

**Comment by vivekverma-arista** [comment]
@judyjoseph there is a cherry pick conflict for msft-202405. Can you look into this?

**Pull request merged**